### PR TITLE
src/example/derivation_3: Add der. for group 3

### PR DESCRIPTION
**Pull request by mrzelak**
Follow convention introduced in derivation_1 and create derivation for group 3.

### CODE REVIEW

File: src/example/derivation_3.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        graph.add_edge(Edge(node_u.handle, node_v.handle, EdgeAttrs('e', False)))
+
+
+    for (node_a, node_b), (node_c, node_d) in zip(it.pairwise(inner_nodes + [inner_nodes[0]]), it.pairwise(outer_nodes + [outer_nodes[0]])):
+        graph.add_q_hyperedge((node_a, node_b, node_c, node_d), EdgeAttrs('q', False))
+
+    return graph
+
+
+class AxesProvider:
```
Comment (macsko) [inline src/example/derivation_3.py:54]: AxesProvider is unused

File: src/example/derivation_3.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        _fig, axes = plt.subplots(nrows=1, ncols=1)
+        return axes
+
+def main():
+    savedir = Path('docs/derivation_1/')
```
Comment (macsko) [inline src/example/derivation_3.py:63]: Should change the path to `docs/derivation_3/`

File: src/example/derivation_3.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+#       P8(),
+#       P2(),
+#       P3(),
+#       P1(),
+#       FixedInput(46),
+#       P8(),
+#       P8(),
+#       P2(),
+#       P3(),
+#       P1()
```
Comment (macsko) [inline src/example/derivation_3.py:101]: Do we need these commented lines?

Comment (mrzelak) [reply src/example/derivation_3.py]: just as every other comment:)

**Review by macsko — APPROVED** [review]

**Review by kkafar — APPROVED** [review]

**Pull request merged**